src/app/components/RequiredSteps.tsx

```diff
@@ -0,0 +1,52 @@
+'use client';
+import { Grid } from '@mui/material';
+import { usePathname } from 'next/navigation';
+import React, { useEffect } from 'react';
+
+import { AddUserDialog } from './dialogs/AddUserDialog';
+import { useAuth } from '../context/AuthProvider';
+import { apiUser } from '../services/endpoints/user.class';
+
+export const RequiredSteps = ({ children }: { children: React.ReactNode }) => {
+  const { user, isAuthenticated } = useAuth();
+  const pathname = usePathname();
+
+  const [isCreateUserDialogOpened, setIsCreateUserDialogOpened] =
+    React.useState(false);
+  const [hasUser, setHasUser] = React.useState(false);
+  const isAnonymous = !isAuthenticated && pathname === '/viewer';
+
+  useEffect(() => {
+    const fetchUser = async (id: string) =>
+      apiUser
+        .getUser(id)
+        .then(({ data }) => {
+          setHasUser(true);
+          return data;
+        })
+        .catch((e) => {
```

Comment:
Hello @yassinedorbozgithub, Or maybe you prefer to log the error instead?
```suggestion
        .catch(() => {
```

Reply:
Hi @amalessid, i think that we can remove e prop like you suggest